precompiles/utils/src/bytes.rs

```diff
@@ -135,7 +135,7 @@ impl<K: Kind, S: Get<u32>> EvmData for BoundedBytesString<K, S> {
         writer.write_pointer(
             EvmDataWriter::new()
                 .write(U256::from(length))
-                .write(value)
+                .write_raw_bytes(&value)
```

Comment:
Note:
We can't use `.write(value)` here because `EvmData` trait defined for `Vec<T>` has a different implementation of `write` method which is not suitable for our encoding of `MultiLocation`.
Hence we will have to use `.write_raw_bytes(&value)` just like MB.